mathics/builtin/graphics.py

```diff
@@ -694,11 +779,41 @@ def apply(self, c1, c2, evaluation, options):
             if not compute:
                 evaluation.message('ColorDistance', 'invdist', distance_function)
                 return
+        elif distance_function.has_form('List', 2):
+            if distance_function.leaves[0].get_string_value() == 'CMC':
+                if distance_function.leaves[1].get_string_value() == 'Acceptability':
+                    compute = lambda c1, c2: _CMC_distance(100*c1.to_color_space('LAB')[:3],
+                                                            100*c2.to_color_space('LAB')[:3], 2, 1)/100
+                elif distance_function.leaves[1].get_string_value() == 'Perceptibility':
+                    compute = ColorDistance._distances.get("CMC")
+                    
+                elif distance_function.leaves[1].has_form('List', 2):
+                    if (isinstance(distance_function.leaves[1].leaves[0], Integer)
+                    and isinstance(distance_function.leaves[1].leaves[1], Integer)):
+                        if (distance_function.leaves[1].leaves[0].get_int_value() > 0
+                        and distance_function.leaves[1].leaves[1].get_int_value() > 0):
+                            lightness = distance_function.leaves[1].leaves[0].get_int_value()
+                            chroma = distance_function.leaves[1].leaves[1].get_int_value()
+                    
+                            compute = lambda c1, c2: _CMC_distance(100*c1.to_color_space('LAB')[:3],
+                                                                    100*c2.to_color_space('LAB')[:3], lightness, chroma)/100
```

Comment:
Looks like there are some unhanded cases here. Maybe a simpler way would be to define `compute = None` at the top and then check if it's still `None` at the end. 

For example:
```
ColorDistance[Red, Blue, DistanceFunction -> {"CMC", {x, y}}]
```
causes a crash on this branch.